ArtOfIllusion/src/artofillusion/CreateCurveTool.java

```diff
@@ -93,12 +101,22 @@ public void drawOverlay(ViewerCanvas view)
       for (int i = 0; i < mesh.from.length; i++)
         view.drawLine(p[mesh.from[i]], p[mesh.to[i]], ViewerCanvas.lineColor);
     }
-    for (int i = 0; i < clickPoint.size(); i++)
-      {
-        Vec3 pos = (Vec3) clickPoint.lastElement();
-        Vec2 screenPos = view.getCamera().getWorldToScreen().timesXY(pos);
-        view.drawBox((int) screenPos.x-HANDLE_SIZE/2, (int) screenPos.y-HANDLE_SIZE/2, HANDLE_SIZE, HANDLE_SIZE, ViewerCanvas.handleColor);
-      }
+    
+    if(clickPoint.isEmpty()) 
+    {
+      return;
+    }
+    int lastPoint = clickPoint.size()-1;
+    
+    Mat4 cameraWorldToScreen = cam.getWorldToScreen();
+    
+    for (int listIndex = 0; listIndex < clickPoint.size(); listIndex++)
+    {
+      Vec2 screenPos = cameraWorldToScreen.timesXY(clickPoint.get(listIndex));        
+      Color handleColor = (listIndex == lastPoint) ? ViewerCanvas.highlightColor : ViewerCanvas.handleColor;
```

Comment:
ViewerCanvas.handleColor should be ViewerCanvas.disabledColor, since the handles cannot be selected or moved.

Otherwise, it looks good. Good catch.